### PR TITLE
JAMES-3539 PushSubscription/set update types

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

**Review by Arsnael — APPROVED** [review]

File: server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/PushSubscriptionSetMethodContract.scala [hunk 1]
```diff
@@ -912,6 +914,245 @@ trait PushSubscriptionSetMethodContract {
...
+           |                        "properties":["types"]
+           |                    }
+           |                }
+           |            },
+           |            "c1"
+           |        ]
+           |    ]
+           |}""".stripMargin)
+  }
+
```
Comment (vttranlina) [inline server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/PushSubscriptionSetMethodContract.scala:1155]: We don't need to check `validated`? If it = `false` then updateType should be rejected?

Comment (chibenwa) [reply server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/PushSubscriptionSetMethodContract.scala:1155]: Where did you get that?

I just re-read the spec and cannot find anything related to this.

Comment (vttranlina) [reply server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/PushSubscriptionSetMethodContract.scala:1155]: When grooming, I remember

Comment (vttranlina) [reply server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/PushSubscriptionSetMethodContract.scala:1155]: Let me check the wrote on the issue

Comment (vttranlina) [reply server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/PushSubscriptionSetMethodContract.scala:1155]: I can't find it, maybe I was wrong, or we talk but did not write it down.
I just remember the key is "when a subscription is not validated, we need to restrict the update and limit properties when get method".

**Comment by chibenwa** [comment]
Do not merge, I suspect rebase conflicts.

**Comment by Arsnael** [comment]
Rebased to solve merging conflicts

**Pull request merged**